src/Search/SearchContext.tsx

```diff
@@ -73,22 +86,21 @@ export const useSearchOnNavBar = ({
 }) => {
   const navigation = useNavigation()
 
-  const [visible, setVisible] = React.useState(false)
-  const {clearSearch} = useSearch()
+  const {search, inputSearchVisible, inputSearchVisibleChanged, clearSearch} = useSearch()
 
   const handleSearchClose = () => {
-    setVisible(false)
+    inputSearchVisibleChanged(false)
     clearSearch()
   }
   const handleGoBack = () => {
     handleSearchClose()
-    navigation.goBack()
+    if (!inputSearchVisible) navigation.goBack()
```

Comment:
We are using the back button for two actions:

- going back to the previous screen
- close the search input